lib/ecto/associations.ex

```diff
@@ -99,6 +99,24 @@ defmodule Ecto.Associations do
   end
 
   @doc """
+  Checks if an association is loaded.
+
+  ## Examples
+
+    post = Repo.get(Post, 1)
+    Ecto.Associations.loaded?(post.comments) # false
+    post = post |> Repo.preload(:comments)
+    Ecto.Associations.loaded?(post.comments) # true
+
+  """
+  def loaded?(association) do
+    case association do
+      %Ecto.Associations.NotLoaded{} -> false
```

Comment:
The problem is that `Ecto.Associations.NotLoaded` is not yet defined so it throws: `Ecto.Associations.NotLoaded.__struct__/0 is undefined, cannot expand struct Ecto.Associations.NotLoaded`

What's the best practice for fixing this issue?